orte/mca/plm/rsh/plm_rsh_module.c

```diff
@@ -709,6 +713,38 @@ static int setup_launch(int *argcptr, char ***argvptr,
     return ORTE_SUCCESS;
 }
 
+/* close all open file descriptors w/ exception of stdin/stdout/stderr */
+static int close_open_file_descriptors(void) {
+    DIR *dir = opendir("/proc/self/fd");
+    if (NULL == dir) {
+        return ORTE_ERR_FILE_OPEN_FAILURE;
+    }
+    struct dirent *files;
+
+    /* grab the fd of the opendir above so we don't close in the 
+     * middle of the scan. */
+    int dir_scan_fd = dirfd(dir);
```

Comment:
I'm not sure `dirfd()` exists outside of Linux, either...?  It doesn't appear to be on MacOS, at least.